### PR TITLE
Add a new mapfiles function

### PR DESCRIPTION
**Pull request by gkotian**
The current `mapbins()` function is a convenient way to map local files into the package, but there is no reason for it to be limited to only mapping binaries. For this reason, a new `mapfiles()` function is introduced.

However, when mapping non-binaries, `VAR.suffix` may not be needed to be appended. So we need to also provide a way to control this.

### CODE REVIEW

File: mkpkg
Comment (gkotian) [inline mkpkg]: This line is quite long. It could be split up as:
```python
    suffix = vars_suffix if (
        'append_suffix' in kwargs and kwargs['append_suffix']
        ) else ''
```
but I'm not sure if that makes it better.

**Comment by leandro-lucarella-sociomantic** [comment]
This should be based on v1.x.x really, because is a new feature.

File: RELEASE_NOTES.md
Comment (leandro-lucarella-sociomantic) [inline RELEASE_NOTES.md]: Could you submit this in a separate PR against v1.8.x?

File: mkpkg
Comment (leandro-lucarella-sociomantic) [inline mkpkg]: When the expression gets a bit complicated another option is just use a classical `if` statement:
```py
suffix = ""
if 'append_suffix' in kwargs and kwargs['append_suffix']:
    suffix = self.pkg.vars['suffix']
```

But this is probably a shorter and more clear alternative:

```py
suffix = kwargs.get('append_suffix', "")
```

:)

File: mkpkg
Comment (leandro-lucarella-sociomantic) [inline mkpkg]: Not a biggie, but in Python the convention is to not leave spaces around the `=` when passing keyword arguments.

**Review by leandro-lucarella-sociomantic — CHANGES_REQUESTED** [review]
I'm not sure if it's worth keeping both really. Specially I don't like the asymmetry in the handling of the suffix in both `map` functions, and if we add the suffix handling to `mapbins` then both functions are identical, right?

I was assuming that binaries will always get the suffix, but then some users expressed the intention of not doing that, which would also make the existent of the specialization less justified.

I wonder.... :thinking:

File: RELEASE_NOTES.md
Comment (gkotian) [reply RELEASE_NOTES.md]: https://github.com/sociomantic-tsunami/makd/pull/58

**Comment by gkotian** [comment]
Thanks for the review, and for the tips on making the code more pythonic.

>I'm not sure if it's worth keeping both really. Specially I don't like the asymmetry in the handling of the suffix in both map functions, and if we add the suffix handling to mapbins then both functions are identical, right?

My main motivation for keeping both functions was to cause zero changes to any code that is already using `FUN.mapbins()`. But otherwise, I also don't think there is much value in keeping both functions.
Since makd's packaging feature is clearly marked as still being a WIP, I think it is reasonable to expect any users of `mapbins()` to make minor adjustments to their code.

I'll update the PR to remove `mapbins()` and only keep `mapfiles()`.

**Comment by gkotian** [comment]
Updated with `FUN.mapbins()` renamed to `FUN.mapfiles()` and with the ability to control whether `VAR.suffix` is appended to the destination file name or not.

**Comment by llucax** [comment]
> Updated with FUN.mapbins() renamed to FUN.mapfiles() and with the ability to control whether VAR.suffix is appended to the destination file name or not.

Oh! Sorry, i didn't mean to remove mapbins completely but just deprecate it and remove it in 2.0.0 to follow semver/Neptune.

**Comment by llucax** [comment]
Also, please don't mix Bugfixes (typos), new features, deprecations and/or breaking changes in the same PR, all of those types of changes usually need to go to specific branches so they can't be all merged as part of the same PR:)

**Comment by llucax** [comment]
So, to summarise: all bug fixes please submit against v1.8.x

The addition of mapfiles and deprecation of mapbins to v1.x.x (will target v1.9.x)

The removal of mapbins to 2.x.x.

Thanks!

**Comment by gkotian** [comment]
>Oh! Sorry, i didn't mean to remove mapbins completely but just deprecate it and remove it in 2.0.0 to follow semver/Neptune.

How can I mark a function as deprecated in Python? There's one approach described in [this stackoverflow answer](http://stackoverflow.com/a/30253848), but I don't know if that's the recommended way.

**Comment by gkotian** [comment]
>Also, please don't mix Bugfixes (typos), new features, deprecations and/or breaking changes in the same PR, all of those types of changes usually need to go to specific branches so they can't be all merged as part of the same PR:)

I guess you're referring to the `Fix typo` commit. That particular typo was introduced in 63f59dc9b4c048affadcd6f6b46f87c95c98c52e which afaict is not in any release. So including that fix in this PR (which targets `v1.x.x`) is correct I think. Can you confirm?

**Comment by gkotian** [comment]
Updated to perform a sort of "soft deprecation" of `FUN.mapbins()`. This means that the release notes specifically states that `FUN.mapbins()` is deprecated and all mentions of it from the README are removed. However if someone still uses it, no deprecation warnings will be seen.

**Comment by leandro-lucarella-sociomantic** [comment]
> How can I mark a function as deprecated in Python? There's one approach described in this stackoverflow answer, but I don't know if that's the recommended way.

I would keep it simple, just `warnf()` when the *deprecated* function is used, that's all.

**Comment by leandro-lucarella-sociomantic** [comment]
> I guess you're referring to the Fix typo commit. That particular typo was introduced in 63f59dc which afaict is not in any release. So including that fix in this PR (which targets v1.x.x) is correct I think. Can you confirm?

You are right, disregard my message.

File: mkpkg
Comment (leandro-lucarella-sociomantic) [inline mkpkg:168]: Here, just add `warnf("FUN.mapbins() is deprecated, please use FUN.mapfiles() instead')`.

**Review by leandro-lucarella-sociomantic — APPROVED** [review]
Once the warning is added, this looks good to me. In the meantime, can other people test this too?

@joseph-wakeling-sociomantic @nemanja-boric-sociomantic maybe? :)

**Comment by leandro-lucarella-sociomantic** [comment]
(needs rebasing too, but better wait until #61 is merged for that, I guess)

**Comment by gkotian** [comment]
Rebased over latest `v1.x.x` and added warning.

**Pull request merged**
